metadata/src/main/java/org/apache/kafka/controller/PartitionChangeBuilder.java

```diff
@@ -311,9 +311,10 @@ private boolean canElectLastKnownLeader() {
                     topicId, partitionId, Arrays.toString(partition.lastKnownElr));
             return false;
         }
-        if (isAcceptableLeader.test(partition.lastKnownElr[0])) {
+        if (!isAcceptableLeader.test(partition.lastKnownElr[0])) {
             log.trace("Try to elect last known leader for {}-{} but last known leader is not alive. last known leader={}",
```

Comment:
Sorry for late reviews. 

It seems this comment is not totally accurate. The `isAcceptableLeader` could reject the leader if the leader, while still alive, has no available directory for the partition.